### PR TITLE
Fix opentelemetry-util-http tests

### LINKED ISSUE #2633: ModuleNotFoundError for opentelemetry.semconv and ImportError for trace in tox -e py39-test-util-http`
### Describe your environment

OS: MacOS with apple chip
Python version: 3.12
Package version: newest

### What happened?

When running tox -e py39-test-util-http, I encounter ModuleNotFoundError and ImportError errors. Specifically, the opentelemetry.semconv module cannot be found, and there is an import error for the trace module.

When I run `tox -e py39-test-util-http`
I will get errors:
```
__________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_capture_custom_headers.py __________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_capture_custom_headers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_capture_custom_headers.py:18: in <module>
    from opentelemetry.util.http import (
.tox/py39-test-util-http/lib/python3.9/site-packages/opentelemetry/util/http/__init__.py:25: in <module>
    from opentelemetry.semconv.trace import SpanAttributes
E   ModuleNotFoundError: No module named 'opentelemetry.semconv'
________________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_http_base.py _________________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_http_base.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_http_base.py:17: in <module>
    from opentelemetry import trace
E   ImportError: cannot import name 'trace' from 'opentelemetry' (unknown location)

```

### Steps to Reproduce

```
$ brew install python

$ python3 -m venv venv

$ source venv/bin/activate

$ pip install tox

$ pip install opentelemetry-api

$ pip install opentelemetry-sdk

$ tox -e py312-test-util-http
```

### Expected Result

the tox will run successfully

### Actual Result

```
__________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_capture_custom_headers.py __________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_capture_custom_headers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_capture_custom_headers.py:18: in <module>
    from opentelemetry.util.http import (
.tox/py39-test-util-http/lib/python3.9/site-packages/opentelemetry/util/http/__init__.py:25: in <module>
    from opentelemetry.semconv.trace import SpanAttributes
E   ModuleNotFoundError: No module named 'opentelemetry.semconv'
________________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_http_base.py _________________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_http_base.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_http_base.py:17: in <module>
    from opentelemetry import trace
E   ImportError: cannot import name 'trace' from 'opentelemetry' (unknown location)
____________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_http_excluded_urls.py ____________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_http_excluded_urls.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_http_excluded_urls.py:18: in <module>
    from opentelemetry.util.http import get_excluded_urls
.tox/py39-test-util-http/lib/python3.9/site-packages/opentelemetry/util/http/__init__.py:25: in <module>
    from opentelemetry.semconv.trace import SpanAttributes
E   ModuleNotFoundError: No module named 'opentelemetry.semconv'
____________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_remove_credentials.py ____________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_remove_credentials.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_remove_credentials.py:3: in <module>
    from opentelemetry.util.http import remove_url_credentials
.tox/py39-test-util-http/lib/python3.9/site-packages/opentelemetry/util/http/__init__.py:25: in <module>
    from opentelemetry.semconv.trace import SpanAttributes
E   ModuleNotFoundError: No module named 'opentelemetry.semconv'
_____________________________________________________________________________ ERROR collecting util/opentelemetry-util-http/tests/test_sanitize_method.py ______________________________________________________________________________
ImportError while importing test module '/Users/lizhihan/PycharmProjects/opentelemetry-python-contrib/util/opentelemetry-util-http/tests/test_sanitize_method.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../miniforge3/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
util/opentelemetry-util-http/tests/test_sanitize_method.py:18: in <module>
    from opentelemetry.util.http import (
.tox/py39-test-util-http/lib/python3.9/site-packages/opentelemetry/util/http/__init__.py:25: in <module>
    from opentelemetry.semconv.trace import SpanAttributes
E   ModuleNotFoundError: No module named 'opentelemetry.semconv'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 5 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================================================================== 5 errors in 0.08s ===========================================================================================================

```

### Additional context

_No response_

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by emdneto**
# Description
It seems opentelemetry-util-http tox commands aren't working properly and it's has been used for several instrumentations. Fix that

Fixes #2633

```shell
$ tox -e py312-test-util-http

========================================================================= warnings summary ==========================================================================
opentelemetry-instrumentation/src/opentelemetry/instrumentation/dependencies.py:4
  /Users/emidio.neto/workspace/emd/github/opentelemetry/opentelemetry-python-contrib/opentelemetry-instrumentation/src/opentelemetry/instrumentation/dependencies.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    from pkg_resources import (

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================== 24 passed, 1 warning in 3.64s ===================================================================
  py312-test-util-http: OK (28.81=setup[0.05]+cmd[5.56,5.44,5.67,5.32,1.70,1.12,3.94] seconds)
  congratulations :) (28.94 seconds)
```

### CODE REVIEW

**Review by Charlie-lizhihan — APPROVED** [review]

File: util/opentelemetry-util-http/test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+importlib-metadata==6.11.0
+iniconfig==2.0.0
+packaging==24.0
+pluggy==1.5.0
+py-cpuinfo==9.0.0
+pytest==7.4.4
+pytest-benchmark==4.0.0
+tomli==2.0.1
+typing_extensions==4.10.0
+-e opentelemetry-instrumentation
```
Comment (lzchen) [inline util/opentelemetry-util-http/test-requirements.txt:12]: I'm not seeing usage of `opentelemetry-instrumentation` libraries anywhere in the tests? Is this a typo?

Comment (emdneto) [reply util/opentelemetry-util-http/test-requirements.txt:12]: Actually is a dependency of the package itself:
If I remove that I get:
```log
.tox/py312-test-util-http/lib/python3.12/site-packages/opentelemetry/util/http/httplib.py:30: in <module>
    from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
E   ModuleNotFoundError: No module named 'opentelemetry.instrumentation'
```

File: tox.ini [hunk 2]
```diff
@@ -761,6 +761,11 @@ commands_pre =
   processor-baggage: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
   processor-baggage: pip install -r {toxinidir}/processor/opentelemetry-processor-baggage/test-requirements.txt
 
+  http: pip install opentelemetry-api@{env:CORE_REPO}\#egg=opentelemetry-api&subdirectory=opentelemetry-api
+  http: pip install opentelemetry-semantic-conventions@{env:CORE_REPO}\#egg=opentelemetry-semantic-conventions&subdirectory=opentelemetry-semantic-conventions
+  http: pip install opentelemetry-sdk@{env:CORE_REPO}\#egg=opentelemetry-sdk&subdirectory=opentelemetry-sdk
```
[the remark this answers is not in the document]
Comment (emdneto) [reply tox.ini:766]: Yes, because of `opentelemetry-test-utils`

**Review by xrmx — APPROVED** [review]
Good catch!

**Review by lzchen — APPROVED** [review]

**Pull request merged**